tools/bundle-build/build.py

```diff
@@ -0,0 +1,22 @@
+import os
+import sys
+import tempfile
+import urllib.request
+from lib.manifest import BuildManifest
+
+if (len(sys.argv) < 2):
+    print("Build an OpenSearch Bundle")
+    print("usage: build.sh /path/to/manifest")
+    exit(1)
+
+with tempfile.TemporaryDirectory() as work_dir:
+    manifest = BuildManifest.from_file(sys.argv[1])
+    build = manifest.build()
+    print(f'Building {build.name()} ({build.arch()})')
+
+    os.chdir(work_dir)
+
+    for component in manifest.components():
+        print(f'=== Building {component.name()} ...')
+        component.checkout()
+        component.build(build.version(), build.arch())
```

Comment:
If i'm reading this correctly, its building in the temp dir but not copying component artifacts into a common output directory.  Each build.sh is currently placing artifacts in <repo-dir>/<component-name>-artifacts.

This should put all components into a specified output directory

Reply:
Also on that note I'd like the output to be something like this.

```
./build
   <min-artifact>
   ./plugins
       <plugin-zips>
   ./libs
```

This makes bundle assembly easier to write.

Not critical for this PR, I can follow up with that change

Reply:
I will update this to pass an output dir reference to each build.sh so that it can place those components directly where they need to go.

Reply:
With https://github.com/opensearch-project/opensearch-build/pull/145/commits/b97225e3921bf5d3dda0e97ebf89e3c4745fea3a you'll get a successful build up to performance-analyzer and this output in `artifacts`:

```
drwxrwxr-x  6 ubuntu ubuntu      4096 Aug  5 23:03 maven
drwxrwxr-x  2 ubuntu ubuntu      4096 Aug  5 23:04 plugins
-rw-rw-r--  1 ubuntu ubuntu 230910212 Aug  5 23:02 opensearch-min-1.0.0-linux-x64.tar.gz
```

... `plugins`:

```
-rw-rw-r-- 1 ubuntu ubuntu 10802657 Aug  5 23:04 opensearch-alerting-1.0.0.0.zip
-rw-rw-r-- 1 ubuntu ubuntu   972506 Aug  5 23:03 opensearch-job-scheduler-1.0.0.0.zip
-rw-rw-r-- 1 ubuntu ubuntu    19084 Aug  5 23:03 opensearch-job-scheduler-sample-extension-1.0.0.0.zip
-rw-rw-r-- 1 ubuntu ubuntu 39048243 Aug  5 23:04 opensearch-security-1.0.0.0.zip
-rw-rw-r-- 1 ubuntu ubuntu 17419492 Aug  5 23:03 opensearch-sql-1.0.0.0.zip
```

... `maven`:

```
drwxrwxr-x  4 ubuntu ubuntu 4096 Aug  5 23:02 common-utils
drwxrwxr-x 34 ubuntu ubuntu 4096 Aug  5 23:02 opensearch
drwxrwxr-x  4 ubuntu ubuntu 4096 Aug  5 23:03 opensearch-job-scheduler
drwxrwxr-x  4 ubuntu ubuntu 4096 Aug  5 23:03 opensearch-job-scheduler-spi
```